### PR TITLE
Process scheduler keeps track of special groups

### LINKED ISSUE #3040: process scheduler does not keep track of special groups
**Describe the bug**
Whenever a process is scheduled, information about user scheduling it is tracked. But we don't keep track of special groups, if any, active for the user at the moment of scheduling. 
In such cases, if scheduled process requires permission(s) related to a special group, it might fail. 

To provide a scenario if an user is granted as administrator by shibboleth, but he / she isn't administrator according to dspace systems, and with administrator grant can schedule a process, when the process is triggered and permissions are re-checked we do not have evidence of previous administrator grant.

### PR DESCRIPTION
**Pull request by corrad82-4s**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #3040

## Description
When a Process is created by a logged in user, having one or more special groups set in his / her session context, uuid of those groups are stored in a table.

In this way, processes needing to check permissions when running, special groups can be added to the context, in order to have a context similar to the one existing when process has been created.

The addition of special groups to context during process runtime is not part of this request. In a way similar to what already done when current user is set in context, Processes needing to perform part of their logics on special groups, can obtain groups list from DSpaceRunnableHandler instance, and added them to their running context.

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* `ProcessService.create` signature updated, accepting a list of special groups
* Process Entity updated with a new groups attribute
* Table representing a one-to-many relation between a process and its related special groups added
* `DSpaceRunnableHandler` interface contract updated with a method returning a list of UUIDs representing special groups related to process being handled by handler.

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 
The quickest way to test this PR is to configure local DSpace7 instance in order to having a special group set at login, with `authentication-password.login.specialgroup `property properly set. Then with a logged in user, schedule a process. Scheduled process will appear in `process` table, and uuid of the special group set at login will appear, together with process' id in new `process2group` table.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Review by abollini — CHANGES_REQUESTED** [review]
hi @corrad82-4s code looks good few remarks in line

File: dspace-api/src/main/java/org/dspace/scripts/Process.java [hunk 1]
```diff
@@ -211,6 +221,14 @@ public Date getCreationTime() {
         return creationTime;
     }
 
+    public List<Group> getGroups() {
+        return groups;
+    }
+
+    public void addGroup(Group group) {
```
[the remark this answers is not in the document]
Comment (corrad82-4s) [reply dspace-api/src/main/java/org/dspace/scripts/Process.java]: Yes, it makes sense. Code has been updated in latest commit

File: dspace-api/src/main/java/org/dspace/scripts/ProcessServiceImpl.java [hunk 2]
```diff
@@ -74,13 +76,22 @@ public class ProcessServiceImpl implements ProcessService {
 
     @Override
     public Process create(Context context, EPerson ePerson, String scriptName,
```
[the remark this answers is not in the document]
Comment (corrad82-4s) [reply dspace-api/src/main/java/org/dspace/scripts/ProcessServiceImpl.java:78]: I have updated already defined javadoc of ProcessService.create method

**Comment by corrad82-4s** [comment]
PR has been added with a commit resolving previously failed automatic builds

**Comment by tdonohue** [comment]
@corrad82-4s : I'm having trouble reviewing this PR as it appears to somehow have a large number of unrelated commits pulled into this PR.  I'm seeing 54 commits here, but it appears only the last 12 or so are applicable to this PR.  Unfortunately though, when I try to review the code it's showing all the changes from the 54 commits.

Could you rebase or cleanup this PR so that it's only containing the commits associated with this work?  If that's not possible, we may need to recreate this PR.  Thanks!

**Comment by corrad82-4s** [comment]
> @corrad82-4s : I'm having trouble reviewing this PR as it appears to somehow have a large number of unrelated commits pulled into this PR. I'm seeing 54 commits here, but it appears only the last 12 or so are applicable to this PR. Unfortunately though, when I try to review the code it's showing all the changes from the 54 commits.
> 
> Could you rebase or cleanup this PR so that it's only containing the commits associated with this work? If that's not possible, we may need to recreate this PR. Thanks!

hi @tdonohue sorry for this mistake. I think something went wrong while realigning this branch with main. If you agree, I can create a new PR and disregard this.

**Comment by tdonohue** [comment]
@corrad82-4s : Feel free to create a replacement PR if this one cannot be cleaned up.

**Comment by corrad82-4s** [comment]
@tdonohue , @abollini : as anticipated, this PR can be disregarded. I have opened a new one: #8297

**Pull request closed without merging**